src/runtimes/node/native_modules/detector.js

```diff
@@ -0,0 +1,16 @@
+// eslint-disable-next-line complexity
+const isNativeModule = ({ binary, dependencies = {}, devDependencies = {}, gypfile }) =>
```

Comment:
I've opted for inlining this check over using [`is-native-module`](https://www.npmjs.com/package/is-native-module) because we also want to look for certain packages in `devDependencies`.

Reply:
> Can we merge dependencies and devDependencies and pass the merged object to is-native-module?

Since this is such a hot function, I'd prefer to avoid the cost of creating a new object each time, if we can help it. The impact should be really minimal, but we're trying to heavily optimise this function.

> We could also submit a PR to is-native-module to check for dev dependencies too?

Sounds good! I'll do that.

Reply:
I'm good with inlining to unblock and using the new `is-native-module` once the PR lands